src/v/archival/ntp_archiver_service.cc

```diff
@@ -2162,9 +2179,11 @@ ss::future<> ntp_archiver::garbage_collect_archive() {
     }
 
     // Drop out if we have no work to do, avoid doing things like the following
-    // manifest flushing unnecessarily.
+    // manifest flushing unnecessarily. This is potentially problematic since,
+    // we've already checked that the clean offset is greater than the start
+    // offset.
     if (objects_to_remove.empty() && manifests_to_remove.empty()) {
-        vlog(_rtclog.debug, "Nothing to remove in archive GC");
+        vlog(_rtclog.warn, "Nothing to remove in archive GC");
```

Comment:
nit: Maybe this should be an error so we will be able to detect this in tests?

Reply:
Sure, I'll do it in a follow up such that we can unblock the rc.